### ISSUE TITLE
Accessibility based cache deletion fails on MIUIv14

### ISSUE THREAD

**Issue by kreicis13** [d4rken-org/sdmaid-se#1521]
[Screenshot_2024-12-29-19-49-23-085_com android settings](https://github.com/user-attachments/assets/af62bc03-af51-4eeb-893f-4ed6e4fa0294)

**Describe the bug**
It's scanning and finds 5 GB of data, and it's takes 20 min to get through all the apps, but at the end nothing was deleted. And  I give accessibility, but still nothing.
**To Reproduce**

**Expected behavior**

**Smartphone (please complete the following information):**
 - Device: Xiaomi T13 Pro
 - OS: [e.g. Android 14up1a]
 - Root status [e.g. Rooted - Magisk v25.2]
 - SD Maid 2/SE version [e.g. v0.6.6-rc6]

**Debug log**
If you can reproduce the bug, record the event in [a debug log](https://github.com/d4rken-org/sdmaid-se/wiki/Bugs#debug-log).

**Screenshots/Video**
If applicable, add screenshots or a video if it helps to explain your problem.

**Additional context**

**Comment by d4rken** [comment]
SD Maid is likely not understanding the screen and it needs to be adjusted for your ROM and Android version. Please provide a debug log. You don't need to record the full 20 min a few min are enough.

**Comment by kreicis13** [comment]
[eu.darken.sdmse_logfile_1735551405703.log](https://github.com/user-attachments/files/18275540/eu.darken.sdmse_logfile_1735551405703.log)

**Comment by kreicis13** [comment]
No updates so far!?

**Comment by d4rken** [comment]
> No updates so far!?

Holidays around xmas and new year are quite busy :wink: 

Clicking fails because the element is disabled:

```java
2024-12-30T09:37:33.808Z  V/SDMSE:Automation:Crawler: crawl():------6: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=false, viewIdResourceName=null, pkgName=com.miui.securitycenter, identity=be7f351
2024-12-30T09:37:33.808Z  V/SDMSE:Automation:Crawler: crawl():-------7: className=android.widget.ImageView, text='null', isClickable=false, isEnabled=false, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_icon, pkgName=com.miui.securitycenter, identity=ee4bb89
2024-12-30T09:37:33.808Z  V/SDMSE:Automation:Crawler: crawl():-------7: className=android.widget.TextView, text='Clear data', isClickable=false, isEnabled=false, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_text, pkgName=com.miui.securitycenter, identity=8dd6690
2024-12-30T09:37:33.811Z  V/SDMSE:Automation:Crawler: Target node found: android.view.accessibility.AccessibilityNodeInfo@256dc3; boundsInParent: Rect(0, 0 - 160, 41); boundsInScreen: Rect(931, 2465 - 1091, 2506); boundsInWindow: Rect(931, 2465 - 1091, 2506); packageName: com.miui.securitycenter; className: android.widget.TextView; text: Clear data; error: null; maxTextLength: -1; stateDescription: null; contentDescription: null; tooltipText: null; containerTitle: null; viewIdResName: com.miui.securitycenter:id/action_menu_item_child_text; uniqueId: null; checkable: false; checked: false; focusable: false; focused: false; selected: false; clickable: false; longClickable: false; contextClickable: false; enabled: false; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_NEXT_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_PREVIOUS_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_SET_SELECTION - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]; isTextSelectable: false
2024-12-30T09:37:33.812Z  V/SDMSE:Automation:Crawler: Target node is className=android.widget.TextView, text='Clear data', isClickable=false, isEnabled=false, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_text, pkgName=com.miui.securitycenter, identity=80533f1
2024-12-30T09:37:33.812Z  V/SDMSE:Automation:Crawler: Mapped node is className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=false, viewIdResourceName=null, pkgName=com.miui.securitycenter, identity=fbf5e57
2024-12-30T09:37:33.812Z  V/SDMSE:Automation:Crawler: Performing action on android.view.accessibility.AccessibilityNodeInfo@256641; boundsInParent: Rect(0, 0 - 366, 203); boundsInScreen: Rect(828, 2367 - 1194, 2570); boundsInWindow: Rect(828, 2367 - 1194, 2570); packageName: com.miui.securitycenter; className: android.widget.LinearLayout; text: null; error: null; maxTextLength: -1; stateDescription: null; contentDescription: Clear data; tooltipText: null; containerTitle: null; viewIdResName: null; uniqueId: null; checkable: false; checked: false; focusable: true; focused: false; selected: false; clickable: true; longClickable: false; contextClickable: false; enabled: false; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_FOCUS - null, AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_NEXT_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_PREVIOUS_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_SET_SELECTION - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]; isTextSelectable: false
2024-12-30T09:37:33.812Z  V/SDMSE:Automation:Crawler:Common: Clicking (isDryRun=false) on className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=false, viewIdResourceName=null, pkgName=com.miui.securitycenter, identity=fbf5e57
2024-12-30T09:37:33.812Z  W/SDMSE:Automation:Crawler: crawl(): Attempt 37 failed on Spec(parent=SDMSE:AppCleaner:Automation:MIUI:Specs, label=CachedCaString("Looking for "Clear data" button (keywords: [Clear data])")):
eu.darken.sdmse.automation.core.errors.DisabledTargetException: Clickable target is disabled.
	at eu.darken.sdmse.main.ui.dashboard.DashboardViewModel$statsItem$2.invokeSuspend(Unknown Source:90)
	at eu.darken.sdmse.main.ui.dashboard.DashboardViewModel$statsItem$2.invoke(Unknown Source:29)
	at eu.darken.sdmse.automation.core.common.StepProcessor.access$doCrawl(Unknown Source:950)
	at eu.darken.sdmse.automation.core.common.StepProcessor$doCrawl$1.invokeSuspend(Unknown Source:13)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:8)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:112)
	at androidx.work.Worker$2.run(Unknown Source:53)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Unknown Source:2)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(Unknown Source:95)
```

Are the apps that fail system apps? From the logs it does not look like it, but just to confirm...

Do you know why the "Clear data" button is disabled?

**Comment by kreicis13** [comment]
I tried to clean cashes in apps settings and it works, so I don't know...

**Comment by d4rken** [comment]
Does it maybe stay disabled for a little bit and then only become clickable?

**Comment by kreicis13** [comment]
Sd maid, goes from app to app, like 10 seconds each.

**Comment by d4rken** [comment]
If you manually enter the app details, do you see the button changing momentarily?

I currently can't reproduce this and need some more help to fix this 🙁

**Comment by ruiasfm** [comment]
I have the same exact problem! I am on a 13t pro with hyper os 2.0.

**Comment by d4rken** [comment]
Does it work if the setting `General > ROM Detection` is set to MIUI?

**Comment by ruiasfm** [comment]
Yes it does!

Thank you. But in case you want to see the debug log I sent you an email
with it.

Rui



A sexta, 17/01/2025, 21:01, Matthias Urhahn ***@***.***>
escreveu:

**Comment by ruiasfm** [comment]
So, the "Operating system detection" option to MIUI doesn't hold. Keeps coming back to automatic every time the app is opened. I have the run in the background turned on and every permission is authorized. 

If you have any ideas to solve this issue kindly tell me. Thank you. 

Rui

**Comment by d4rken** [comment]
> Yes it does!

That's good to know, I think I can come up with a fix then without a test-device.

> So, the "Operating system detection" option to MIUI doesn't hold. Keeps coming back to automatic every time the app is opened.

Good catch, I never noticed this. Will be fixed in the next update.

**Comment by d4rken** [comment]
@ruiasfm Please continue in #1538 , based on the original log of @kreicis13  and yours, it's two different issues. @kreicis13 does not have HyperOS 2.0

**Comment by ruiasfm** [comment]
Do you need something else from me? Need some more information?

Rui

A domingo, 19/01/2025, 11:14, Matthias Urhahn ***@***.***>
escreveu:
